R/age_model_class.R

```diff
@@ -0,0 +1,222 @@
+#' Defines an age-structured simple SEIR model.
+#' Slots of this class are as follows
+#'
+#' @slot name character representing name of model
+#' @slot output_names names of the compartments which are used by the
+#'     model.
+#' @slot parameter_names names of the initial conditions and parameters
+#'     which are used by the model.
+#' @slot parameters named list containing parameters of the model.
+#'     Initial values for each compartment, S0, E0, I0, R0.
+#'     Other parameters: a, b, c which represent the rates of changes
+#'     between the compartments.
+#' @slot n_age_categories number of age categories.
+#'
+#' @import deSolve
+#' @import glue
+#' @import ggplot2
+
+setClass('age_model',
+         #slots
+         slots = c(
+           name = 'character',
+           output_names = 'list',
+           parameter_names = 'list',
+           parameters = 'list',
+           n_age_categories = 'numeric'
+         ),
+
+         #prototypes for the slots, automatically set output and param
+         #names
+         prototype = list(
+           name = NA_character_,
+           output_names = list('S', 'E', 'I', 'R', 'Incidence'),
+           parameter_names = list('S0', 'E0', 'I0', 'R0', 'a', 'b', 'c'),
+           parameters = vector(mode = "list", length = 7),
+           n_age_categories = NA_real_
+
+         )
+)
+
+#' @describeIn Retrieves parameters for an age-structured simple SEIR model.
+#'
+#' @param object An object of the class age_model.
+setGeneric('get_parameters', function(object) standardGeneric('get_parameters'))
```

Comment:
I'd be tempted to change it from `get_parameters` to `transmission_parameters`. You might also make a setter and a getter for it as described here: https://adv-r.hadley.nz/s4.html

Reply:
we use get_parameters as it  gets the initial population fractions as well as transmission parameters but more than happy to change if you think it's best. Alternatively could separate out get functions for initial fractions and transmission parameters (something like initials() & transimssion_parameters()).

Reply:
Up to you: I'd be tempted to separate them as there will be quite a few parameters, and it'd be good to avoid a long list. But pretty each about this point.